File: lesson_22_java/src/main/java/com/lesson22java/controllers/ProductController.java
package com.lesson22java.controllers;

import com.lesson22java.entities.Product;
import com.lesson22java.services.ProductService;
import lombok.AllArgsConstructor;
import org.springframework.web.bind.annotation.*;


@RestController
@AllArgsConstructor
@RequestMapping("/product")
public class ProductController {
    private ProductService productService;

    @GetMapping("/create")
    public Product create(@ModelAttribute Product product) {
        return productService.create(product);
    }

    @GetMapping("/getAll")
    public Iterable<Product> getAll() {
        return productService.getAll();
    }

    @GetMapping("/getOne/{id}")
    public Product getOne(@PathVariable Long id) {
        return productService.getOne(id);
    }

    @GetMapping("/edit")
    public Product edit(@ModelAttribute Product product) {
        return productService.edit(product);
    }

    @GetMapping("/delete/{id}")
    public void delete(@PathVariable Long id) {
        productService.delete(id);
    }

}

File: lesson_22/src/main/kotlin/com/lesson22/controllers/ProductController.kt
package com.lesson22.controllers

import com.lesson22.entities.Product
import com.lesson22.services.ProductService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.ModelAttribute
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/products")
class ProductController (val productService: ProductService) {
    @GetMapping("/get-all")
    fun getAll() = productService.getAll()

    @GetMapping("/create")
    fun create(@ModelAttribute product: Product) = productService.create(product)

    @GetMapping("/delete/{id}")
    fun delete(@PathVariable id: Long) = productService.delete(id)

    @GetMapping("/edit")
    fun edit(@ModelAttribute product: Product) = productService.edit(product)
}